File: TPTD3/Article.java
public class Article {
    int reference;
    String libelle;
    double prix;
    public Article(int r,String l,double p){
        this.reference = r;
        this.libelle = l;
        this.prix = p;
    }
    public int getReference() {
        return reference;
    }
    public void setReference(int reference) {
        this.reference = reference;
    }
    public String getLibelle() {
        return libelle;
    }
    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }
    public double getPrix() {
        return prix;
    }
    public void setPrix(double prix) {
        this.prix = prix;
    }
    @Override
    public String toString() {
        return "Article [reference=" + reference + ", libelle=" + libelle + ", prix=" + prix + "]";
    }
}

File: TPTD4/Article.py
class Article:

    def __init__(self,reference,libelle,prix) -> None:
        self._reference = reference
        self._libelle = libelle
        self._prix = prix

    def __repr__(self) -> str:
        res ="Article %d nommé %s coûte %r"%(self._reference,self._libelle,self._prix)
        return res
    @property
    def getRef(self):
        return self._reference
    
    @property
    def getLibelle(self):
        return self._libelle
    
    @property
    def getPrix(self):
        return self._prix
    
    def setRef(self,ref):
        self._reference = ref
    
    def setLibelle(self,l):
        self._libelle = l
    
    def setPrix(self,p):
        self._prix = p